### PR TITLE
refactor(consul): code clearing

### PR DESCRIPTION
**Pull request by ardasnturk**
- simplified `RefreshControl` with `usePullToRefetch`
  to avoid code duplication.
- added "is" to the beginning of `useState(boolean)`
  expressions to understand that the variable
  is boolean.
- sorted imports and variables alphabetically
  to improve readability
- added `defaultValues` props to input to
  eliminate warn error
- deleted props names written twice to
  simplify code
- replaced `val` in 'onSubmit(val)' with a
  more understandable name
- updated propTypes to `navigation.object`
  as all necessary information is available in `navigation.object`
- removed component statement to improve
  readability of file directory
- moved normalize import from `react-native-
  elements` to `config` due to style bug
- renamed `onRefresh` props to `refetch` to
  eliminate name confusion

SVA-403

### CODE REVIEW

File: src/components/consul/detail/ConsulCommentListItem.js [hunk 1]
```diff
@@ -36,14 +36,14 @@ const deleteCommentAlert = (onDelete) =>
...
-  const [showReply, setShowReply] = useState(false);
+export const ConsulCommentListItem = ({ commentItem, refetch, replyList, navigation }) => {
   const [isLoading, setIsLoading] = useState(false);
+  const [isShowReply, setIsShowReply] = useState(false);
+  const [isShowResponse, setIsShowResponse] = useState(false);
 
   const { control, handleSubmit, reset } = useForm({
     defaultValues: {
-      comment: ''
+      comment: null
```
Comment (donni106) [inline src/components/consul/detail/ConsulCommentListItem.js:46]: why was this changed to null?

File: src/components/consul/detail/ConsulCommentListItem.js [hunk 2]
```diff
@@ -36,14 +36,14 @@ const deleteCommentAlert = (onDelete) =>
 
 /* eslint-disable complexity */
 /* NOTE: we need to check a lot for presence, so this is that complex */
-export const ConsulCommentListItem = ({ commentItem, onRefresh, replyList, navigation }) => {
-  const [showResponse, setShowResponse] = useState(false);
-  const [showReply, setShowReply] = useState(false);
+export const ConsulCommentListItem = ({ commentItem, refetch, replyList, navigation }) => {
   const [isLoading, setIsLoading] = useState(false);
+  const [isShowReply, setIsShowReply] = useState(false);
+  const [isShowResponse, setIsShowResponse] = useState(false);
```
Comment (donni106) [inline src/components/consul/detail/ConsulCommentListItem.js:42]: `show` would be enough to indicate a boolean. it does not have to be `is` all the time. see it as natural language when you ask for something.

1. is something loading? -> `isLoading`?
2. should replies be shown? -> `showReply`?
3. should responses be shown? -> `showResponses`?

questions for conditions could be understand very fine i guess. with `is` and `show` at the same time, it seems duplicated for me and i have difficulties to make a natural sentence. you know what i mean?

File: src/components/consul/detail/ConsulQuestionsList.js [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
-import { ConsulQuestionsListItem } from './ConsulQuestionsListItem';
 import { ConsulQuestionsDescriptionListItem } from './ConsulQuestionsDescriptionListItem';
+import { ConsulQuestionsListItem } from './ConsulQuestionsListItem';
 
-const text = texts.consul;
-
-export const ConsulQuestionsList = ({ data, onRefresh, token, disabled }) => {
-  const [userAnswer, setAnswerUser] = useState(false);
+export const ConsulQuestionsList = ({ data, refetch, token, disabled }) => {
+  const [isUserAnswer, setIsAnswerUser] = useState(false);
```
Comment (donni106) [inline src/components/consul/detail/ConsulQuestionsList.js:13]: `isUserAnswer` does not match `setIsAnswerUser` as expected

File: src/components/consul/detail/ConsulVideo.js [hunk 4]
```diff
@@ -24,14 +24,14 @@ export const ConsulVideoComponent = ({ videoUrl }) => {
...
+ConsulVideo.propTypes = {
   videoUrl: PropTypes.string
 };
 
 const styles = StyleSheet.create({
   videoView: {
     flex: 1,
-    width: '100%',
-    height: 250
+    height: 250,
```
Comment (donni106) [inline src/components/consul/detail/ConsulVideo.js:34]: nothing new in code here, but with seeing it i wonder if `250` always works for different devices and orientations? can you have a look in different situations please.

File: src/components/consul/detail/ConsulVoting.js [hunk 5]
```diff
@@ -80,18 +80,18 @@ export const ConsulVotingComponent = ({ votesData, onRefresh, id }) => {
...
-  onRefresh: PropTypes.func
-};
-
 const styles = StyleSheet.create({
+  icon: {
+    paddingHorizontal: 10
+  },
   iconButton: {
     alignItems: 'center',
     marginHorizontal: 10
```
Comment (donni106) [inline src/components/consul/detail/ConsulVoting.js:89]: do we need normalization here? or is it ok not having it in this particular place.

File: src/components/screens/consul/Debates/DebateDetail.js [hunk 6]
```diff
@@ -53,14 +53,14 @@ export const DebateDetail = ({ data, onRefresh, route, navigation }) => {
...
-      if (val) return setUserId(JSON.parse(val).id);
+    getConsulUser().then((userInfo) => {
+      if (userInfo) return setUserId(JSON.parse(userInfo).id);
     });
   }, []);
 
   const { control, handleSubmit, reset } = useForm({
     defaultValues: {
-      comment: ''
+      comment: null
```
Comment (donni106) [inline src/components/screens/consul/Debates/DebateDetail.js:63]: same here as asked before, why changed to `null`?

File: src/components/screens/consul/Debates/NewDebate.js [hunk 7]
```diff
@@ -42,15 +42,15 @@ export const NewDebate = ({ navigation, data, query }) => {
...
       id: data?.id,
       attributes: {
         translationsAttributes: {
-          title: val.title,
-          description: val.description
+          title: newDebateData.title,
+          description: newDebateData.description
         },
-        tagList: val.tagList,
+        tagList: newDebateData.tagList,
```
Comment (donni106) [inline src/components/screens/consul/Debates/NewDebate.js:53]: these should be secured with conditional chaining i guess to avoid crashes if `newDebateData` is undefined.

File: src/components/screens/consul/Polls/PollDetail.js [hunk 8]
```diff
@@ -53,7 +53,7 @@ export const PollDetail = ({ data, onRefresh, route, navigation }) => {
 
   const { control, handleSubmit, reset } = useForm({
     defaultValues: {
-      comment: ''
+      comment: null
```
Comment (donni106) [inline src/components/screens/consul/Polls/PollDetail.js:56]: again this as twice before

File: src/components/screens/consul/Proposals/NewProposal.js [hunk 9]
```diff
@@ -65,11 +72,11 @@ export const NewProposal = ({ navigation, data, query }) => {
...
-      title: data?.title,
-      description: data?.description,
-      tagList: data?.tagList?.toString(),
-      summary: data?.summary,
-      videoUrl: data?.videoUrl
+      title: data?.title ? data?.title : null,
+      description: data?.description ? data?.description : null,
+      tagList: data?.tagList ? data?.tagList?.toString() : null,
+      summary: data?.summary ? data?.summary : null,
+      videoUrl: data?.videoUrl ? data?.videoUrl : null
```
Comment (donni106) [inline src/components/screens/consul/Proposals/NewProposal.js:79]: again, why have you extended the conditions here to return `null`s instead of `undefined`s?

File: src/components/screens/consul/Proposals/ProposalDetail.js [hunk 10]
```diff
@@ -69,14 +69,14 @@ export const ProposalDetail = ({ data, onRefresh, route, navigation }) => {
...
-      if (val) return setUserId(JSON.parse(val).id);
+    getConsulUser().then((userInfo) => {
+      if (userInfo) return setUserId(JSON.parse(userInfo).id);
     });
   }, []);
 
   const { control, handleSubmit, reset } = useForm({
     defaultValues: {
-      comment: ''
+      comment: null
```
Comment (donni106) [inline src/components/screens/consul/Proposals/ProposalDetail.js:79]: again this :)

File: src/components/screens/consul/Proposals/ProposalDetail.js [hunk 11]
```diff
@@ -246,15 +246,13 @@ export const ProposalDetail = ({ data, onRefresh, route, navigation }) => {
 };
 /* eslint-enable complexity */
 
+const styles = StyleSheet.create({
+  imageContainerStyle: { alignSelf: 'center' }
```
Comment (donni106) [inline src/components/screens/consul/Proposals/ProposalDetail.js:250]: i think style sheets are never one lined across the whole code base. could you please adjust it here as well?

File: src/screens/consul/ConsulIndexScreen.js [hunk 12]
```diff
@@ -53,13 +53,13 @@ const showRegistrationFailAlert = (navigation) =>
...
-  const [refreshing, setRefreshing] = useState(false);
-  const [sortingType, setSortingType] = useState(INITIAL_TOP_SORTING);
   const [filterType, setFilterType] = useState(INITIAL_TOP_FILTERING_FOR_POLLS);
   const [queryVariables, setQueryVariables] = useState(route.params?.queryVariables ?? {});
+  const [sortingType, setSortingType] = useState(INITIAL_TOP_SORTING);
   const bookmarkable = route.params?.bookmarkable;
-  const query = route.params?.query ?? '';
   const extraQuery = route.params?.extraQuery ?? '';
+  const query = route.params?.query ?? '';
+  let type = null;
```
Comment (donni106) [inline src/screens/consul/ConsulIndexScreen.js:62]: this `type` is used for two purposes, right? is there a specific reason using it globally? it is very not clear when reading `type = null`.
i saw the first for sorting list data, where `type` is used to store a selected one. couldn't it be a local `const selectedSortingType` for example?
and second usage is for filtering, where `type` is also used to store a selected one. couldn't it be a local `const selectedFilterType` for example?
this way there would no `type` variable, which has not much meaning and is used for multiple purposes.

File: src/screens/consul/LoginRegister/ConsulLoginScreen.js [hunk 13]
```diff
@@ -34,42 +35,51 @@ const showLoginFailAlert = () =>
...
 
   const {
     control,
     formState: { errors },
     handleSubmit
-  } = useForm();
+  } = useForm({
+    defaultValues: {
+      email: null,
+      password: null
```
Comment (donni106) [inline src/screens/consul/LoginRegister/ConsulLoginScreen.js:48]: does react hook form like `null` more than `''`?

**Review by donni106 — CHANGES_REQUESTED** [review]
i like this pull request a lot! with a few answers and adjustments this will be a big improvement for the code base.

File: src/components/consul/detail/ConsulCommentListItem.js (hunk 1 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulCommentListItem.js:46]: i thought `null` was more accurate both in terms of readability and in terms of disk space.

File: src/components/consul/detail/ConsulCommentListItem.js (hunk 2 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulCommentListItem.js:42]: Yes, I understand better. Thank you.

File: src/components/consul/detail/ConsulCommentListItem.js (hunk 1 above)
Comment (donni106) [reply src/components/consul/detail/ConsulCommentListItem.js:46]: does the react-hook-form docs say something about that?

File: src/components/consul/detail/ConsulVoting.js (hunk 5 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulVoting.js:89]: I have tested. Better to use. It is not understandable unless you are very careful with devices of different sizes. But still there is a slight difference.

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulVideo.js:34]: I tested. There is no big difference in devices of different sizes. How do you think I can do this?

File: src/components/screens/consul/Debates/DebateDetail.js (hunk 6 above)
Comment (ardasnturk) [reply src/components/screens/consul/Debates/DebateDetail.js:63]: i thought null was more accurate both in terms of readability and in terms of disk space.

File: src/components/screens/consul/Debates/NewDebate.js [hunk 14]
```diff
@@ -21,17 +21,17 @@ const graphqlErr = (err) => Alert.alert('Hinweis', err);
...
     formState: { errors },
     handleSubmit
   } = useForm({
     defaultValues: {
-      title: data?.title,
-      description: data?.description,
-      tagList: data?.tagList?.toString()
+      title: data?.title ? data?.title : null,
+      description: data?.description ? data?.description : null,
+      tagList: data?.tagList ? data?.tagList?.toString() : null
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/screens/consul/Debates/NewDebate.js:34]: React-Hook-Form was giving warn error for `defaultValues`. To clear this error.

File: src/screens/consul/LoginRegister/ConsulLoginScreen.js (hunk 13 above)
Comment (ardasnturk) [reply src/screens/consul/LoginRegister/ConsulLoginScreen.js:48]: The documentation states that you can use null or empty strings.

`defaultValues` Note: It is encouraged that you set a defaultValue for all inputs to non-undefined such as the empty string or null.

[React-Hook-Form v6](https://react-hook-form.com/v6/api/)

Comment (donni106) [reply src/screens/consul/LoginRegister/ConsulLoginScreen.js:48]: ah ok. good.
what do you think about using empty string for string fields and null for others?

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (donni106) [reply src/components/consul/detail/ConsulVideo.js:34]: i thought for landscape for example the video will be very wide but still 250 in height. but maybe our orientation wrapper already saves us for having such "weird" renderings.

File: src/screens/consul/LoginRegister/ConsulLoginScreen.js (hunk 13 above)
Comment (ardasnturk) [reply src/screens/consul/LoginRegister/ConsulLoginScreen.js:48]: All code needs to stay in the same structure. I will update my code with empty string.

File: src/screens/consul/ConsulIndexScreen.js (hunk 12 above)
Comment (ardasnturk) [reply src/screens/consul/ConsulIndexScreen.js:62]: Good suggestion. I wanted to do 2 different jobs with a single variable.

File: src/screens/consul/LoginRegister/ConsulLoginScreen.js (hunk 13 above)
Comment (donni106) [reply src/screens/consul/LoginRegister/ConsulLoginScreen.js:48]: do you mean updating empty strings just for string values or for everything? i think we should not have default values with empty strings, if the values at the end are numbers for examples.

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulVideo.js:34]: Most probably. Should I change this?

File: src/components/consul/detail/ConsulCommentListItem.js (hunk 1 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulCommentListItem.js]: https://github.com/ikuseiGmbH/smart-village-app-app/pull/521#discussion_r862853106

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (donni106) [reply src/components/consul/detail/ConsulVideo.js:34]: did you encounter problems now that need to be changed? which one?

File: src/screens/consul/ConsulIndexScreen.js (hunk 12 above)
Comment (ardasnturk) [reply src/screens/consul/ConsulIndexScreen.js]: 8d1121aeab4ff33f5e37bc5f2568faec55c0b11f

File: src/components/consul/detail/ConsulVoting.js (hunk 5 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulVoting.js]: 8d1121aeab4ff33f5e37bc5f2568faec55c0b11f

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (ardasnturk) [reply src/components/consul/detail/ConsulVideo.js:34]: i didn't see any problem.

File: src/components/screens/consul/Proposals/NewProposal.js [hunk 15]
```diff
@@ -120,13 +120,13 @@ export const NewProposal = ({ navigation, data, query }) => {
...
-          title: newProposalData.title,
-          summary: newProposalData.summary,
-          description: newProposalData.description
+          title: newProposalData?.title,
+          summary: newProposalData?.summary,
+          description: newProposalData?.description
         },
-        tagList: newProposalData.tagList,
+        tagList: newProposalData?.tagList,
         termsOfService: termsOfService,
```
Comment (donni106) [inline src/components/screens/consul/Proposals/NewProposal.js:128]: `termsOfService: termsOfService` -> `termsOfService` ;)

Comment (donni106) [reply src/components/screens/consul/Proposals/NewProposal.js:128]: and this is also a boolean, i just see: `const [termsOfService, setTermsOfService] = useState(data?.termsOfService ?? false);`

so please change it to `hasAcceptedTermsOfService` or something like that more understandable

File: src/screens/consul/ConsulIndexScreen.js [hunk 16]
```diff
@@ -59,7 +59,8 @@ export const ConsulIndexScreen = ({ navigation, route }) => {
   const bookmarkable = route.params?.bookmarkable;
   const extraQuery = route.params?.extraQuery ?? '';
   const query = route.params?.query ?? '';
-  let type = null;
+  const selectedSortingType = sortingType.find((data) => data.selected);
+  const selectedFilterType = filterType.find((data) => data.selected);
```
Comment (donni106) [inline src/screens/consul/ConsulIndexScreen.js:63]: please place them in the corresponding methods. we do not need them globally. it is only needed in the local methods (list data callback for sorting and effect for filtering)

File: src/components/consul/detail/ConsulVideo.js (hunk 4 above)
Comment (donni106) [reply src/components/consul/detail/ConsulVideo.js:34]: ok, then we do not need to change anything.

File: src/screens/consul/ConsulIndexScreen.js (hunk 16 above)
Comment (ardasnturk) [reply src/screens/consul/ConsulIndexScreen.js]: 7a3a9dacaabe27e7b1291911444f7d2b05906983

File: src/components/screens/consul/Proposals/NewProposal.js (hunk 15 above)
Comment (ardasnturk) [reply src/components/screens/consul/Proposals/NewProposal.js]: 7a3a9dacaabe27e7b1291911444f7d2b05906983

**Review by donni106 — APPROVED** [review]
